builder/vsphere/common/step_reattach_cdrom.go

```diff
@@ -46,43 +47,50 @@ func (s *StepReattachCDRom) Run(_ context.Context, state multistep.StateBag) mul
 		return multistep.ActionHalt
 	}
 
-	// Keep the CD-ROM devices on the image without any attached media.
 	ui.Say("Reattaching CD-ROM devices...")
-	// Remove all CD-ROM devices from the image.
-	err = vm.RemoveCdroms()
-	if err != nil {
-		state.Put("error", fmt.Errorf("error removing cdrom prior to reattaching: %v", err))
-		return multistep.ActionHalt
-	}
-	// If the CD-ROM device type is SATA, find the SATA controller.
-	if s.CDRomConfig.CdromType == "sata" {
-		if _, err := vm.FindSATAController(); err == driver.ErrNoSataController {
-			ui.Say("Adding SATA controller...")
-			// If a SATA controller is not found, add it.
-			if err := vm.AddSATAController(); err != nil {
-				state.Put("error", fmt.Errorf("error adding sata controller: %v", err))
-				return multistep.ActionHalt
-			}
+	var nActableCdroms int = ReattachCDRom - len(s.CDRomConfig.ISOPaths)
+	if nActableCdroms < 0 {
```

Comment:
Correct. To clarify:

>  we want to reattach 2 cdroms but have 4 attached len(s.CDRomConfig.ISOPaths) this logic will only remove 2 since they will no longer be needed

…and the remaining two will have cdroms ejected 

> If we have 4 attached and want reattach 4 we remove none to essentially reuse attached CDROMs

…and all 4 of them have cdroms ejected